RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -68,45 +68,42 @@ public AnalyzerPluginGenerator(INuGetPackageHandler packageHandler, SonarQube.Pl
             this.logger = logger;
         }
 
-        public bool Generate(NuGetReference analyzerRef, string language, string sqaleFilePath, string outputDirectory)
+        public bool Generate(ProcessedArgs args)
         {
-            // sqale file path is optional
-            if (analyzerRef == null)
-            {
-                throw new ArgumentNullException("analyzeRef");
-            }
-            if (string.IsNullOrWhiteSpace(language))
-            {
-                throw new ArgumentNullException("language");
-            }
-            SupportedLanguages.ThrowIfNotSupported(language);
-            if (string.IsNullOrWhiteSpace(outputDirectory))
+            if (args == null)
             {
-                throw new ArgumentNullException("outputDirectory");
+                throw new ArgumentNullException("args");
             }
 
-            IPackage package = TryGetPackage(analyzerRef);
+            IPackage package = this.packageHandler.FetchPackage(args.PackageId, args.PackageVersion);
             if (package == null)
             {
                 return false;
             }
 
             string packageDir = this.packageHandler.GetLocalPackageRootDirectory(package);
-            IEnumerable<DiagnosticAnalyzer> analyzers = GetAnalyzers(packageDir, this.packageHandler.LocalCacheRoot, language);
-
+            IEnumerable<DiagnosticAnalyzer> analyzers = GetAnalyzers(packageDir, this.packageHandler.LocalCacheRoot, args.Language);
```

Comment:
Changed the order of processing: no point in asking users to accept licenses if the package doesn't contain any analyers